[plugin-html] Allow to use XPath in HTML steps (#2494)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-html.adoc

:locator-types: `CSS selector` or `XPath`

= HTML Plugin

The plugin provides the ability to work with HTML documents: data extraction and validation.

== Installation

.build.gradle
[source,gradle,subs="attributes+"]
----
implementation(group: 'org.vividus', name: 'vividus-plugin-html', version: '{current-version}')
----

== Locators

HTML locators should be used in order to interact with HTML document elements. The following locators are supported:

> - https://jsoup.org/cookbook/extracting-data/selector-syntax[`CSS selector`

Reviewer: ```suggestion
- https://jsoup.org/cookbook/extracting-data/selector-syntax[`CSS selector`]
```

Then element found by XPath `//h3` in HTML `${response}` contains text `111122`
----

=== Save attribute value

Saves the attribute value into the variable.

[source,gherkin]
----
When I save `$attributeName` attribute value of element found by $locatorType `$locator` in HTML `$html` to $scopes variable `$variableName`
----

- `$attributeName` - The name of the attribute.
- `$locatorType` - {locator-types}.
- `$locator` - See <<_locator>>.
- `$html` - The HTML document to find element.
- `$scopes` - xref:commons:variables.adoc#_scopes[The comma-separated set of the variables scopes].
- `$variableName` - The name of the variable to save the attribute value.

.Validate paragraph title
[source,gherkin]
----
When I initialize the story variable `html` with value
`
<html>
  <head>
    <title>Page Title</title>
    <script>//<![CDATA[Here comes the data//]]></script>
  </head>
  <body>
    <h1>This is a Heading</h1>
    <p title="paragraph">This is a paragraph.</p>
  </body>
</html>
`
When I save `title` attribute value of element found by CSS selector `p` in HTML `${html}` to scenario variable `title`
Then `${title}` is = `paragraph`
----

=== Save data/text of element

> Saves the data or the text of element. Where the *data* is characters between the start-tag and end-tag of the element and the *text* is the textual content of the element visible by user.

Reviewer: ```suggestion
Saves the data or the text of element. Where the *data* is characters between the start-tag and end-tag of the element and the *text* is the textual content of the element  without any inner element.
```
